### ISSUE TITLE
When Autopilot enabled, main loop should set Visual metronome counter

### ISSUE THREAD

**Issue by brennansingular** [SingularSound/openbbm#132]
The main loop of a part should tell the BB visual metronome how long the main loop is, and the visual metronome should count the measures. Currently, nothing is shown to count measures elapsed.

For example in part one, the BeatBuddy would count 8 small bars on the visual metronome
![Screenshot (41)](https://user-images.githubusercontent.com/62663388/94608322-45c18a00-026b-11eb-8379-ca254f4b125a.png)

**Comment by brennansingular** [comment]
The biggest question is how would we handle UI for parts with a lot of measures (32)/(64), should we consider an input to multiply part length? Say the main loop will be 8 measures, but you set it to multiply from there (x2, x3, x4 etc). This would be a way to create a clean method for handling long song parts and still having the visual feedback of the part's main loop.

Another option could be to find a UI method that show's multiplicative counting, maybe a secondary measure counter abover the measure counter that is counting "sets" of 8. Is this making sense? 

Let me know your thoughts @GoranRista, @DavidPackouz , @iajrz

**Comment by GoranRista** [comment]
Currently, there is no visual indicator for section lengths in the BB Manager. 
I wouldn't obsess too much about this. Remember that this is not a musical chart that the performer has to clearly see in order to read their parts note by note. It is merely an indication of roughly where they are in a section. I can't imagine anyone intently staring at the BB while they are playing so that they know where they are. I think just keeping a single bar that shows progress (as it has been), should be sufficient. If you see that the bar is barely moving, you'll know that the section has a lot of bars. If you know the song well, you'll match with the BB perfectly. If not, the BB may not be of much help :)

**Comment by iajrz** [comment]
I agree with Goran's comment. In particular, the BBM is reflecting the behavior of the BB, so we should think about these things from the BB and mirror whatever makes sense to mirror in the BBM.

**Comment by brennansingular** [comment]
@GoranRista @iajrz I'm sorry if I was not clear, I'm saying this should be on the BB not the BBM

**Comment by brennansingular** [comment]
BB currently only counts the measures of the midi files used in the main loop, not the main loop full length

**Comment by GoranRista** [comment]
@brennansingular Ok. Got you. In the case of AP, the progress bar should display the full length of the programmed part, rather than the length of the loaded midi file. That would be the only useful scenario since most loop midi files are 1 to 2 bars long.
Once the user restarts the count, the progress bar should also restart.

**Comment by DavidPackouz** [comment]
I agree with @GoranRista 

This was actually in my specification on Basecamp @bacol89

**Comment by iajrz** [comment]
Eh - also, this is the repo for the openbbm, if it's BB strictly, it should exist as an issue in the BB firmware repo.

**Comment by brennansingular** [comment]
@iajrz 
@bacol89 and I have agreed to put all BBM related BB issues here, this seems like a coordinated issue between the two, if you'd like to do otherwise we can do that
